enhancements/operator-hub-check.md

```diff
@@ -0,0 +1,186 @@
+
+#### New Criteria valid which will be checked against to a specific Kubernetes version
+
+For the checks which should be done accordingly for some specific version(s) it will use the Kubernetes version which has been provided to the validator via flag or via the bundle operator configuration. For example, if we run:
+
+```sh
+ operator-sdk bundle validate ./bundle \
+     --select-optional name=operatorhub \
+     --optional-values="k8s=1.22"
+```
+
+Then, it means that the check will validate the operator bundle configuration accordingly to the Kubernetes version informed where it is intend to be published to. In this case, the validator will consider the Kubernetes version `=1.22`. 
+
+If the `--optional-values="k8s=1.22"` not be used then, we will use the `minKubeVersion` in the CSV to do the checks. So, the criteria is `>=minKubeVersion`. 
+
+By last, if the version is not provided then we should consider as described above the operator bundle is intend to work well in any Kubernetes version.
+
+**CRD and Webook APIS deprecated and unsupported versions**
+
+Note that in these cases, users should be informed that the `apiextensions/v1beta1` and that the `admissionregistration.k8s.io/v1beta1` was deprecated in Kubernetes `1.16` and will/was removed in `1.22`.
+
+- To check if the operator bundle is using a CRD API version which is deprecated or not supported we will need to see if the CRD manifests in the bundle are using `apiVersion: apiextensions.k8s.io/v1beta1`. (See [here](https://github.com/operator-framework/operator-sdk/blob/v1.4.2/testdata/go/v2/memcached-operator/bundle/manifests/cache.example.com_memcacheds.yaml#L1)). Note that we need to check all CRD(s) added on the bundle.
+
+- To check if the operator bundle is using the Webhook API version which is deprecated we will need to see if the CSV has the spec `webhookdefinitions.admissionReviewVersions.v1beta1`. (See [here](https://github.com/operator-framework/operator-sdk/blob/v1.4.2/testdata/go/v2/memcached-operator/bundle/manifests/memcached-operator.clusterserviceversion.yaml#L203-L205)
+
+The error raised will be accordingly with the Kubernetes version that we have, see:
+
+| Conditional |Type |
+| ------ | ----- |
+| operator bundle supports k8s => `1.22` and uses `apiextensions/v1beta1` ÒR `admissionregistration.k8s.io/v1beta1` | error |
+| operator bundle supports k8s `=> 1.16 <= 1.22` and uses `apiextensions/v1beta1` ÒR `admissionregistration.k8s.io/v1beta1` | warning |
+
+**IMPORTANT** 
+
+> Kubernetes version `1.22` was not released until now which means that the deprecated APIs still supported for all clusters. Also, at the same way none OpenShift version which is using Kubernetes API `1.22` was released. Then, we cannot raise errors until it happens for the scenarios where we are checking the version from `minKubeVersion` or assuming all possible versions when it is not provided.
```

Comment:
we can still raise the errors.  we know what will happen when 1.22 is released.  we don't need to wait for it to be released to change the validation logic.  If you are validating against 1.22 (either because that's what your minkubeversion says, or because that's what someone passed on the command line) then it's an error to use v1beta1 apis.

Reply:
If  `minKubeVersion` >= 1.22 yes we can it is true. My concern here is: if `minKubeVersion` == nil scenario good catcher.